src/components/app.cjsx

```diff
@@ -16,12 +19,11 @@ module.exports = React.createClass
     HistoryLocation.removeChangeListener(@storeHistory)
 
   storeInitial: ->
-    if History.length is 1
-      path = HistoryLocation.getCurrentPath()
-      TransitionActions.load({path})
+    @storeHistory(path: HistoryLocation.getCurrentPath())
 
   storeHistory: (locationChangeEvent) ->
-    TransitionActions.load(locationChangeEvent)
+    if DestinationHelper.shouldRememberRoute(locationChangeEvent, @context.router)
+      TransitionActions.load(locationChangeEvent)
 
```

Comment:
Ideally TransitionActions would perform this check itself.  It's unable to though since you can't pass the router into the actions.  Flux clones incoming params which strips all the methods.